src/main/java/fi/helsinki/opintoni/security/SAMLContextProviderReverseProxy.java

```diff
@@ -39,22 +39,19 @@ protected void populateGenericContext(HttpServletRequest request,
         super.populateGenericContext(new ReverseProxyRequestWrapper(request), response, context);
     }
 
-    private static class ReverseProxyRequestWrapper extends HttpServletRequestWrapper {
+    public static class ReverseProxyRequestWrapper extends HttpServletRequestWrapper {
 
         private ReverseProxyRequestWrapper(HttpServletRequest request) {
             super(request);
         }
 
         @Override
         public StringBuffer getRequestURL() {
-            StringBuffer sb = new StringBuffer();
-            sb.append("https://").append(getServerName());
-            sb.append(getContextPath());
-            sb.append(getServletPath());
-            if (getPathInfo() != null) {
-                sb.append(getPathInfo());
+            StringBuffer original = super.getRequestURL();
+            if (original.indexOf("http:") == 0) {
```

Comment:
Voisi käyttää `String.startsWith`ia, ja alempana korvauksessa `String.replaceFirst`iä

Reply:
Toi on alunperin StringBuffer, ei kai sitä kannata muuttaa Strigiksi vaan että oi käyttää noita metodeja?

Reply:
Totta. Katsoin ensin että kyseessä olisi String. Ok siis näin.